app/controllers/employment/OtherPaymentsController.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.employment
+
+import config.{AppConfig, ErrorHandler}
+import controllers.employment.routes.CheckEmploymentDetailsController
+import controllers.predicates.{AuthorisedAction, InYearAction}
+import forms.YesNoForm
+import models.mongo.EmploymentCYAModel
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents}
+import services.EmploymentSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.SessionHelper
+import views.html.employment.OtherPaymentsView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class OtherPaymentsController @Inject()(implicit val cc: MessagesControllerComponents,
+                                        authAction: AuthorisedAction,
+                                        inYearAction: InYearAction,
+                                        otherPaymentsView: OtherPaymentsView,
+                                        appConfig: AppConfig,
+                                        employmentSessionService: EmploymentSessionService,
+                                        errorHandler: ErrorHandler,
+                                        ec: ExecutionContext) extends FrontendController(cc) with I18nSupport with SessionHelper {
+
+  def show(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+      val yesNoForm: Form[Boolean] = YesNoForm.yesNoForm(
+        missingInputError = s"employment.other-payments.errors.noRadioSelected.${if (user.isAgent) "agent" else "individual"}"
+      )
+      Ok(otherPaymentsView(yesNoForm, taxYear, employmentId))
```

Comment:
think this line is left over from the demo, also didnt u wanna make the form reusable?

Reply:
removed